contracts/token/ERC1155/ERC1155.sol

```diff
@@ -369,6 +367,21 @@ contract ERC1155 is Context, ERC165, IERC1155, IERC1155MetadataURI {
         emit TransferBatch(operator, account, address(0), ids, amounts);
     }
 
+    /**
+     * @dev Approve `operator` to operate on all of `owner` tokens
+     *
+     * Emits a {ApprovalForAll} event.
+     */
+    function _setApprovalForAll(
+        address owner,
+        address operator,
+        bool approved
+    ) internal virtual {
+        // TODO: add sanity checks?
```

Comment:
What checks did you have in mind?

Reply:
possibly `require(owner != operator)` ... but I was not really sure

Reply:
I agree with adding that require statement. We should move the one in the external function into the internal function.

Note: This PR is also pending a changelog entry.